src/lib/storage/mvcc_data.cpp

```diff
@@ -26,10 +26,10 @@ void MvccData::shrink() {
   end_cids.shrink_to_fit();
 }
 
-void MvccData::grow_by(size_t delta, TransactionID transaction_id) {
+void MvccData::grow_by(size_t delta, CommitID begin_commit_id) {
   _size += delta;
-  tids.grow_to_at_least(_size, transaction_id);
-  begin_cids.grow_to_at_least(_size, MAX_COMMIT_ID);
+  tids.grow_to_at_least(_size);
```

Comment:
I don't get why this would be faster than `tids.grow_to_at_least(_size, TransactionID{0]);`....

Reply:
[Here](https://github.com/01org/tbb/blob/a51a90bc609bb73db8ea13841b5cf7aa4344d4a9/include/tbb/concurrent_vector.h#L780), either `initialize_array` or `initialize_array_by` get called. Next, `init()` is called [with or without an argument](https://github.com/01org/tbb/blob/2ace525889b0c3de9c90da943fac9259220ef35f/include/tbb/concurrent_vector.h#L1110). Without an argument, it's a default initialization, with an argument, it's a copy construction.

Reply:
Mmh, that might mean that the TIDs are uninitialized though (and were before). I tried moving the non-synchronized code here - let's see what the benchmarks say.

Reply:
Nevermind. Of course, `T()` is zero-initialized.

Reply:
Not so sure anymore. `std::atomic` leaves the default-constructed object in an invalid state. With synchronization deactivated, initialization is not noticeable, so I rather set the tids manually in every case.